selene-lib/src/rules/duplicate_keys.rs

```diff
@@ -0,0 +1,181 @@
+use crate::ast_util::range;
+
+use super::*;
+use std::{collections::HashMap, convert::Infallible};
+
+use full_moon::{
+    ast::{self, Ast},
+    tokenizer,
+    visitors::Visitor,
+};
+
+pub struct DuplicateKeysLint;
+
+impl Rule for DuplicateKeysLint {
+    type Config = ();
+    type Error = Infallible;
+
+    fn new(_: Self::Config) -> Result<Self, Self::Error> {
+        Ok(DuplicateKeysLint)
+    }
+
+    fn pass(&self, ast: &Ast, _: &Context) -> Vec<Diagnostic> {
+        let mut visitor = DuplicateKeysVisitor {
+            duplicates: Vec::new(),
+        };
+
+        visitor.visit_ast(&ast);
+
+        visitor
+            .duplicates
+            .iter()
+            .map(|duplicate| {
+                Diagnostic::new_complete(
+                    "duplicate_keys",
+                    format!("key `{}` is already declared", duplicate.name),
+                    Label::new(duplicate.position),
+                    Vec::new(),
+                    vec![Label::new_with_message(
+                        duplicate.original_declaration,
+                        format!("`{}` originally declared here", duplicate.name),
+                    )],
+                )
+            })
+            .collect()
+    }
+
+    fn severity(&self) -> Severity {
+        Severity::Warning
+    }
+
+    fn rule_type(&self) -> RuleType {
+        RuleType::Complexity
```

Comment:
```suggestion
        RuleType::Correctness
```

This is guaranteed to be wrong, especially with the guarantees the lint has.

Reply:
Whoops, forgot to change this